storage_controller/src/tenant_shard.rs

```diff
@@ -463,6 +467,7 @@ impl TenantShard {
             last_error: Arc::default(),
             pending_compute_notification: false,
             scheduling_policy: ShardSchedulingPolicy::default(),
+            preferred_az_id: None,
```

Comment:
nit: "preferred_az_id" is a bit of a mouthful.

Could just be "az", or "home_az", something like that?

Reply:
`home_az` implies a stronger constraint to me. I don't mind, so if you feel strongly will update.